expression/builtin.go

```diff
@@ -29,17 +27,18 @@ import (
 
 // baseBuiltinFunc will be contained in every struct that implement builtinFunc interface.
 type baseBuiltinFunc struct {
-	args      []Expression
-	argValues []types.Datum
-	ctx       context.Context
-	self      builtinFunc
+	args          []Expression
+	argValues     []types.Datum
+	ctx           context.Context
+	deterministic bool
 }
 
 func newBaseBuiltinFunc(args []Expression, ctx context.Context) baseBuiltinFunc {
 	return baseBuiltinFunc{
-		args:      args,
-		argValues: make([]types.Datum, len(args)),
-		ctx:       ctx,
+		args:          args,
+		argValues:     make([]types.Datum, len(args)),
+		ctx:           ctx,
+		deterministic: true,
```

Comment:
I suggest use `false` as the default value, if someone else use `&baseBuiltinFunc{}` instead of `newBaseBuiltinFunc`, it might introduce bug. So `false` as default value makes the code more robust in Go.